prog/vm/github_runner.rb

```diff
@@ -62,6 +62,31 @@ def pick_vm
     vm_st.subject
   end
 
+  def find_closest_docker_mirror_server
+    compatible_servers = DockerRegistryMirrorServer.all.select { |server| server.vm.arch == vm.arch }
+    return nil if compatible_servers.empty?
+
+    same_vm_host_server = nil
+    same_data_center_server = nil
+    same_region_server = nil
+    fallback_server = nil
+
+    compatible_servers.each do |server|
+      if server.vm.vm_host.id == vm.vm_host_id
+        same_vm_host_server = server
+        break
+      elsif server.vm.vm_host.data_center == vm.vm_host.data_center
+        same_data_center_server = server
+      elsif server.vm.vm_host.data_center&.split("-")&.first == vm.vm_host.data_center&.split("-")&.first
```

Comment:
nit: Since it has too many `&`, it might be difficult to achieve 100% coverage.

```suggestion
      elsif server.vm.vm_host.data_center.to_s[/^[^-]+/] == vm.vm_host.data_center.to_s[/^[^-]+/]
```